climbing_stairs/climbing_stairs.py

```diff
@@ -2,9 +2,43 @@
 
 import sys
 
-def climbing_stairs(n, cache=None):
-  pass 
+# Iterative
+# def climbing_stairs(n, cache=None):
+#   n_over_3 = n // 3
+#   remainder = n % 3
 
+#   if n < 3:
+#     if n == 2:
+#       return 2
+#     if n == 1:
+#       return 1
+#     else:
+#       return 0
+
+#   if remainder == 2:
+#     return n_over_3 * 4 + 2
+#   elif remainder == 1:
+#     return n_over_3 * 4 + 1
+#   elif remainder == 0:
+#     return  n_over_3 * 4
+
+#Recursive (too slow)
+# def climbing_stairs(n, cache=None):
+#   if n < 0:
+#     return 0
+#   elif n == 0 or n == 1:
+#     return 1
+
+#   return climbing_stairs(n-1) + climbing_stairs(n-2) + climbing_stairs(n-3)   
```

Comment:
You can use memoization/caching to make this faster. This would involve saving a result of a `climbing_stairs(n)` to a list or dictionary, then having future recursive calls check the cache first if the answer is in there already.